Client/Frontend/AuthenticationManager/AuthenticationSettingsViewController.swift

```diff
@@ -232,31 +289,55 @@ class AuthenticationSettingsViewController: SettingsTableViewController {
-                    title: NSAttributedString.tableRowTitle(
-                        NSLocalizedString("Use Touch ID", tableName:  "AuthenticationManager", comment: "List section title for when to use Touch ID")
-                    ),
-                    navigationController: self.navigationController,
-                    delegate: nil,
-                    enabled: true,
-                    touchIDSuccess: { [unowned self] in
-                        self.touchIDAuthenticationSucceeded()
-                    },
-                    touchIDFallback: { [unowned self] in
-                        self.fallbackOnTouchIDFailure()
-                    }
-                )
+        let requirePasscodeSectionChildren: [Setting] = [RequirePasscodeSetting(settings: self)]
+        let authenticationSection = SettingSection(title: authenticationSectionTitle, children: [
+            AuthenticationForPrivateBrowsingSetting(
+                title: NSAttributedString.tableRowTitle(
+                    NSLocalizedString("Private Browsing", tableName:  "AuthenticationManager", comment: "List section title for when to use authentication for private browsing")
+                ),
+                navigationController: self.navigationController,
+                delegate: nil,
+                enabled: true,
+                touchIDSuccess: { touchIDSetting in
+                    touchIDSetting.toggleTouchID(enabled: false)
+                },
+                touchIDFallback: { [unowned self] touchIDSetting in
+                    AppAuthenticator.presentPasscodeAuthentication(self.navigationController,
+                        success: {
+                            touchIDSetting.toggleTouchID(enabled: false)
+                            self.navigationController?.dismissViewControllerAnimated(true, completion: nil)
+                        },
+                        cancel: nil
+                    )
+                }
+            ),
+            AuthenticationForLoginsSetting(
+                title: NSAttributedString.tableRowTitle(
+                    NSLocalizedString("Logins", tableName:  "AuthenticationManager", comment: "List section title for when to use Touch ID for logins")
+                ),
+                navigationController: self.navigationController,
+                delegate: nil,
+                enabled: true,
+                touchIDSuccess: { touchIDSetting in
+                    touchIDSetting.toggleTouchID(enabled: false)
+                },
+                touchIDFallback: { [unowned self] touchIDSetting in
+                    AppAuthenticator.presentPasscodeAuthentication(self.navigationController,
+                        success: { 
+                            touchIDSetting.toggleTouchID(enabled: false)
+                            self.navigationController?.dismissViewControllerAnimated(true, completion: nil)
+                        },
+                        cancel: nil
+                    )
+                }
```

Comment:
Nit: I think we can clean this up a bit. Defining nested callbacks inside callbacks inside instantiating a setting inside an array seems a bit messy.

Since they rely on the `navigationController`, maybe define private helper functions that spits out each setting. Something like,

```swift

private func privateBrowserSetting(navigationController: UINavigationController) {
    return AuthenticationForPrivateBrowsingSetting(
                title: NSAttributedString.tableRowTitle(
                    NSLocalizedString("Private Browsing", tableName:  "AuthenticationManager", comment: "List section title for when to use authentication for private browsing")
                ),
                navigationController: self.navigationController,
                delegate: nil,
                enabled: true,
                touchIDSuccess: { touchIDSetting in
                    touchIDSetting.toggleTouchID(enabled: false)
                },
                touchIDFallback: { [unowned self] touchIDSetting in
                    AppAuthenticator.presentPasscodeAuthentication(self.navigationController,
                        success: {
                            touchIDSetting.toggleTouchID(enabled: false)
                            self.navigationController?.dismissViewControllerAnimated(true, completion: nil)
                        },
                        cancel: nil
                    )
                }
            )
}
```

then,

```swift
let authenticationSection = SettingSection(title: authenticationSectionTitle, children: [
    privateBrowserSetting(self.navigationController),
    ...
])
```